### ISSUE TITLE
howto configure warnings for 8.6

### ISSUE THREAD

**Issue by hendriktews** [ProofGeneral/PG#140]
I open this issue to foster the discussion about how to configure warnings in 8.6. If you have an opinion, please leave a comment.

* The mechanism to configure warnings should work for both, background compilation and coqtop invocation.
* We could insert an Set Warnings command before we process the scripting buffer in coqtop.
* We probably should support to configure warnings, because if background compilation fails, one might see a lot of warnings before the error.
* There seems to be a hierarchy in the warnings, but there is no documentation about it.
* It would be nice to have a list of all warnings and the ones that are enabled by default, but I cannot find either.
* It would surely be impressive to have radio buttons for all the warnings, but do we need this?
* Is there a way to configure warnings in _CoqProject that works with other tools?

**Comment by Matafou** [comment]
coq_makefile accepts option "-arg foo" which pass option foo to coqc and coqtop, and IIRC I I implemented -arg recognition for pg. So I guess it should already work. I can't test right now but I will later today. There may be issues with options with complex syntax.

**Comment by cpitclaudel** [comment]
@Matafou Indeed, this could work; it would be nice to have a configuration system consistent with CoqIDE.
That being said, _CoqProject is so horrendous that this could be another reason to push for a better format.

**Comment by Matafou** [comment]
Someone is recoding coq_makefile currently it may be time to list what you don't like with _CoqProject.

**Comment by cpitclaudel** [comment]
I saw Enrico's email.  I'm hoping to chat about this at POPL.  My issue with _CoqProject is the lack of semantics (It's just a bunch of command line arguments, but the syntax isn't specified, as far as I can tell)

**Comment by cpitclaudel** [comment]
Maybe a more precise characterization is that the original format was nice and simple (sequences of `-optname optarg1 … optargn`), but now that you can add paths you run into problems with quotes and escaping.

**Comment by hendriktews** [comment]
Here is the list of warnings in 8.6 (relying on the pattern CWarnings.create). I am not sure, a menu with 24 submenus and 71 tick boxes in total is useful, although it would be a nice little hack to generate the necessary emacs lisp code automatically...

24 categories with 71 warnings in total

- category automation: 3 warnings
  * cast-in-pattern
  * non-imported-hint
  * polymorphic-hint

- category bytecode-compiler: 1 warnings
  * bytecode-compiler-failed

- category declmode: 2 warnings
  * declmode-insufficient-justification
  * declmode-missing-case

- category deprecated: 14 warnings
  * deprecated-appcontext
  * deprecated-arguments-scope
  * deprecated-arguments-syntax
  * deprecated-eqn-syntax
  * deprecated-implicit-arguments
  * deprecated-include-type
  * deprecated-info-tactical
  * deprecated-inputstate
  * deprecated-local-syntax
  * deprecated-nested-proofs
  * deprecated-option
  * deprecated-outputstate
  * firstorder-deprecated-syntax
  * require-in-module

- category extraction: 8 warnings
  * extraction-ambiguous-name
  * extraction-axiom-to-realize
  * extraction-inside-module
  * extraction-logical-axiom
  * extraction-opaque-accessed
  * extraction-opaque-as-axiom
  * extraction-remaining-implicit
  * extraction-reserved-identifier

- category filesystem: 7 warnings
  * ambiguous-file-name
  * cannot-open-dir
  * cannot-open-path
  * cannot-remove-file
  * cannot-use-directory
  * file-no-extension
  * path-not-found

- category fixpoints: 1 warnings
  * non-full-mutual

- category funind: 3 warnings
  * funind-cannot-build-inversion
  * funind-cannot-define-graph
  * funind-cannot-define-principle

- category implicits: 2 warnings
  * definition-not-visible
  * implicits-in-term

- category loadpath: 1 warnings
  * overriding-logical-loadpath

- category ltac: 3 warnings
  * meta-collision
  * profile-backtracking
  * variable-collision

- category native-compiler: 3 warnings
  * native-compiler-disabled
  * native-compiler-failed
  * native-compute-disabled

- category numbers: 1 warnings
  * large-nat

- category option: 1 warnings
  * unknown-option

- category parsing: 8 warnings
  * comment-terminator-in-string
  * invalid-identifier
  * non-reversible-notation
  * notation-bound-to-variable
  * notation-overridden
  * skip-spaces-curly
  * unrecognized-unicode
  * unusable-identifier

- category pedantic: 1 warnings
  * plural-command

- category record: 1 warnings
  * non-primitive-record

- category records: 1 warnings
  * cannot-define-projection

- category require: 1 warnings
  * several-object-files

- category schemes: 1 warnings
  * cannot-build-congruence

- category scope: 1 warnings
  * local-declaration

- category tactics: 2 warnings
  * simpl-unfolding-modifiers
  * unused-intro-pattern

- category typechecker: 3 warnings
  * projection-no-head-constant
  * redundant-canonical-projection
  * uniform-inheritance

- category vernacular: 2 warnings
  * arguments-assert
  * let-as-axiom
